patch/0013-hwmon-pmbus-shrink-code-and-remove-pmbus_do_remove.patch

```diff
@@ -0,0 +1,511 @@
+ identify function if supported. Must only be called from device probe
+ function.
+ 
+-::
+-
+-  void pmbus_do_remove(struct i2c_client *client);
+-
+-Execute driver remove function. Similar to standard driver remove function.
+-
+ ::
+ 
+   const struct pmbus_driver_info
+diff --git a/Documentation/hwmon/pmbus.rst b/Documentation/hwmon/pmbus.rst
+index fb3ad67dedc1..c44f14115413 100644
+--- a/Documentation/hwmon/pmbus.rst
++++ b/Documentation/hwmon/pmbus.rst
+@@ -148,11 +148,6 @@ Emerson DS1200 power modules might look as follows::
+ 	return pmbus_do_probe(client, &ds1200_info);
+   }
+ 
+-  static int ds1200_remove(struct i2c_client *client)
+-  {
+-	return pmbus_do_remove(client);
+-  }
+-
+   static const struct i2c_device_id ds1200_id[] = {
+ 	{"ds1200", 0},
+ 	{}
+@@ -166,7 +161,6 @@ Emerson DS1200 power modules might look as follows::
+ 		   .name = "ds1200",
+ 		   },
+ 	.probe_new = ds1200_probe,
+-	.remove = ds1200_remove,
+ 	.id_table = ds1200_id,
+   };
+ 
+diff --git a/drivers/hwmon/pmbus/adm1266.c b/drivers/hwmon/pmbus/adm1266.c
+index c7b373ba92f2..4d2e4ddcfbfd 100644
+--- a/drivers/hwmon/pmbus/adm1266.c
++++ b/drivers/hwmon/pmbus/adm1266.c
+@@ -502,7 +502,6 @@ static struct i2c_driver adm1266_driver = {
+ 		   .of_match_table = adm1266_of_match,
+ 		  },
+ 	.probe_new = adm1266_probe,
+-	.remove = pmbus_do_remove,
+ 	.id_table = adm1266_id,
+ };
+ 
+diff --git a/drivers/hwmon/pmbus/adm1275.c b/drivers/hwmon/pmbus/adm1275.c
+index e7997f37b266..38a6515b0763 100644
+--- a/drivers/hwmon/pmbus/adm1275.c
++++ b/drivers/hwmon/pmbus/adm1275.c
+@@ -797,7 +797,6 @@ static struct i2c_driver adm1275_driver = {
+ 		   .name = "adm1275",
+ 		   },
+ 	.probe_new = adm1275_probe,
+-	.remove = pmbus_do_remove,
+ 	.id_table = adm1275_id,
+ };
+ 
+diff --git a/drivers/hwmon/pmbus/bel-pfe.c b/drivers/hwmon/pmbus/bel-pfe.c
+index 2c5b853d6c7f..aed7542d7ce5 100644
+--- a/drivers/hwmon/pmbus/bel-pfe.c
++++ b/drivers/hwmon/pmbus/bel-pfe.c
+@@ -121,7 +121,6 @@ static struct i2c_driver pfe_pmbus_driver = {
+ 		   .name = "bel-pfe",
+ 	},
+ 	.probe_new = pfe_pmbus_probe,
+-	.remove = pmbus_do_remove,
+ 	.id_table = pfe_device_id,
+ };
+ 
+diff --git a/drivers/hwmon/pmbus/dni_dps460.c b/drivers/hwmon/pmbus/dni_dps460.c
+index 4e96a430bbe9..0d6fc448d27f 100644
+--- a/drivers/hwmon/pmbus/dni_dps460.c
++++ b/drivers/hwmon/pmbus/dni_dps460.c
+@@ -375,7 +375,6 @@ static int dni_dps460_remove(struct i2c_client *client)
+ 	sysfs_remove_group(&client->dev.kobj, &dni_dps460_attr_grp);
+ 	if (data->info)
+ 		kfree(data->info);
+-	pmbus_do_remove(client);
+ 	return 0;
+ }
+ 
+diff --git a/drivers/hwmon/pmbus/dps1900.c b/drivers/hwmon/pmbus/dps1900.c
+index decbd07c6d6e..c7400d4f1bfa 100644
+--- a/drivers/hwmon/pmbus/dps1900.c
++++ b/drivers/hwmon/pmbus/dps1900.c
+@@ -151,7 +151,6 @@ static struct i2c_driver dps1900_driver = {
+ 		.name = "dps1900",
+ 	},
+ 	.probe = dps1900_probe,
+-	.remove = pmbus_do_remove,
```

Comment:
Another case of an upstream driver addition that was missed by the patch. Corrected here.